### PR TITLE
LibLine: Handle <return>s in incomplete data correctly

### PR DESCRIPTION
**Pull request by alimpfard**
Previously, we would concatenate all the commands together:
```
> sleep 5
echo well
echo hello
echo friends

> echo wellecho helloecho friends
```

Thanks, @bugaevc

### CODE REVIEW

File: Libraries/LibLine/Editor.cpp [hunk 1]
```diff
@@ -222,8 +222,14 @@ String Editor::get_line(const String& prompt)
         }
 
         Utf8View input_view { StringView { m_incomplete_data.data(), valid_bytes } };
+        size_t index_in_view = 0;
 
         for (auto ch : input_view) {
+            if (m_finish)
```
Comment (bugaevc) [inline Libraries/LibLine/Editor.cpp:228]: `ch` is actually `codepoint` isn't it?

File: Libraries/LibLine/Editor.cpp [hunk 2]
```diff
@@ -222,8 +222,14 @@ String Editor::get_line(const String& prompt)
         }
 
         Utf8View input_view { StringView { m_incomplete_data.data(), valid_bytes } };
+        size_t index_in_view = 0;
```
Comment (bugaevc) [inline Libraries/LibLine/Editor.cpp:225]: And the index is — the index, counting in codepoints (not bytes)?

Comment (alimpfard) [reply Libraries/LibLine/Editor.cpp:225]: well yes, it's `index_in_view`, you're thinking of a better name?

Comment (bugaevc) [reply Libraries/LibLine/Editor.cpp:225]: `number_of_codepoints`? idk, I'm not familiar with LibLine internals, just bringing up that `Utf8View` doesn't actually support indexing, so "index in view" doesn't tell me much

Comment (alimpfard) [reply Libraries/LibLine/Editor.cpp:225]: does `consumed_codepoints` sound ok?

Comment (bugaevc) [reply Libraries/LibLine/Editor.cpp:225]: Yeah

**Pull request merged**